### PR TITLE
headscale: 0.21.0 → 0.22.1

### PR DESCRIPTION
**Pull request by johnrichardrinehart**
###### Description of changes

Just bump the version.

###### Things done

Bumped it!

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@kradalby @Misterio77 @ghuntley

### CODE REVIEW

**Review by NKJe — APPROVED** [review]
LGTM

File: pkgs/servers/headscale/default.nix [hunk 1]
```diff
@@ -6,16 +6,16 @@
...
   src = fetchFromGitHub {
     owner = "juanfont";
     repo = "headscale";
     rev = "v${version}";
-    hash = "sha256-Y4fTCEKK7iRbfijQAvYgXWVa/6TlPikXnqyBI8b990s=";
+    sha256 = "sha256-6T4wWuhikanoQGGjVvNJak5yvgcEfhGtOmfLc2xKmms=";
   };
 
-  vendorHash = "sha256-R183PDeAUnNwNV8iE3b22S5hGPJG8aZQGdENGqcPCw8=";
+  vendorSha256 = "sha256-+JxS4Q6rTpdBwms2nkVDY/Kluv2qu2T0BaOIjfeX85M=";
```
Comment (06kellyjac) [inline pkgs/servers/headscale/default.nix:18]: These should have stayed hash and vendorHash

**Review by 06kellyjac — COMMENTED** [review]
Thanks for the PR
Some notes for the future:

The title and commit should not have a v prefix & just match the nix package version

And the attached comment

File: pkgs/servers/headscale/default.nix (hunk 1 above)
Comment (johnrichardrinehart) [reply pkgs/servers/headscale/default.nix:18]: Deal. I'll put up a PR to revert the attributes. Is `*sha256` legacy compared to `*hash` such that both are supported?

Comment (06kellyjac) [reply pkgs/servers/headscale/default.nix:18]: Yeah SRI identified hashes have been growing in use in nixpkgs. Youd have to be on quite an old copy of nix to not support them AFAIK

Comment (johnrichardrinehart) [reply pkgs/servers/headscale/default.nix:18]: @06kellyjac I pushed a PR to fix this but it was commented, there, that it's still either-or, for now: https://github.com/NixOS/nixpkgs/pull/227783/commits/84be28f36ac4ce0632df8ff2cd91822b234360db .

**Pull request merged**